### PR TITLE
Create test users

### LINKED ISSUE #319: Create test users
When developing locally, it would be ideal to have a db seed to set up a few test users.

Firebase emulators don't implement a way to `auth:import` & `auth:export` yet, for the `auth` data. This means we have to handle this manually with some `ts` code.

At the moment, I create two accounts from the front-end side (only when using emulator), and I could change related test users data in `onUserCreate` hook.

A better way would be to setup the db through the `Admin SDK`.

### PR DESCRIPTION
**Pull request by MattiaPrimavera**
Fix #319

### CODE REVIEW

File: functions/src/triggers/onUserCreation.ts [hunk 1]
```diff
@@ -3,11 +3,49 @@ import * as jsonWebToken from 'jsonwebtoken'
...
+        sentEmails: 0,
+    },
+})
+
+const getUserData = (user: admin.auth.UserRecord) => {
+    let userData
+    if (process.env.FIREBASE_AUTH_EMULATOR_HOST) {
+        switch (user.email) {
+            case TEST_ACCOUNTS.paidUser.email:
+                userData = makeUserData(true, false)
```
Comment (HugoGresse) [inline functions/src/triggers/onUserCreation.ts:29]: return directly rather than mutating

File: functions/src/db/constants.ts [hunk 2]
```diff
@@ -7,3 +7,18 @@ export const COLLECTIONS = {
 export const SUBSCRIPTIONS = {
     manual: 'manual',
 }
+
+export const TEST_ACCOUNTS = {
```
Comment (HugoGresse) [inline functions/src/db/constants.ts:11]: duplicated file from back and front :(

File: functions/src/triggers/onUserCreation.ts (hunk 1 above)
Comment (MattiaPrimavera) [reply functions/src/triggers/onUserCreation.ts:29]: clean code: a function should rather have a single success exit point

Comment (HugoGresse) [reply functions/src/triggers/onUserCreation.ts:29]: yes but what about immutability?

File: functions/src/db/constants.ts (hunk 2 above)
Comment (MattiaPrimavera) [reply functions/src/db/constants.ts:11]: #352

File: functions/src/triggers/onUserCreation.ts (hunk 1 above)
Comment (MattiaPrimavera) [reply functions/src/triggers/onUserCreation.ts:29]: I guess it is a compromise to accept in order to increase readability in this case ;)

Comment (HugoGresse) [reply functions/src/triggers/onUserCreation.ts:29]: what about reducer? they all return directly within a switch

**Review by HugoGresse — APPROVED** [review]

File: functions/src/triggers/onUserCreation.ts (hunk 1 above)
Comment (MattiaPrimavera) [reply functions/src/triggers/onUserCreation.ts:29]: good point. We may consider [refactoring that](https://medium.com/better-programming/write-cleaner-reducers-1c433a94a166)

**Pull request merged**